harvester_e2e_tests/integration/test_volumes.py

```diff
@@ -72,3 +74,68 @@ def _wait_for_volume_ready():
         timeout=request.config.getoption('--wait-timeout'))
     assert success, 'Timed out while waiting for volume to be ready.'
     return pvc_json
+
+
+@pytest.mark.volumes
+@pytest.mark.p1
+def test_create_volume_backing_image(api_client, unique_name, opensuse_image,
+                                     wait_timeout, request):
+    """
+    1. Create a new image from URL
+    2. Check that it is created succesffully.
+    3. Create a new volume with the image_id of the previous image
+    4. Check that the new image is created
+    5. Delete image and volume
+    """
+
+    code, image_data = api_client.images.create_by_url(unique_name, opensuse_image.url)
+    if code != 201:
+        raise AssertionError(
+            f"Failed to create image {unique_name} from URL got\n"
+            f"Creation got {code} with {image_data}"
+        )
+
+    assert 201 == code, (code, image_data)
+
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+    while endtime > datetime.now():
+        code, data = api_client.images.get(unique_name)
+        image_conds = data.get('status', {}).get('conditions', [])
+        if len(image_conds) > 0:
+            break
+        sleep(3)
```

Comment:
I would suggest to create a general variable for the wait time instead of hardcoded 3

Reply:
That's an excellent point. I'll do that.

Reply:
@khushboo-rancher I think the while loop already explain the sleep in here is targeting to snooze a bit to do another check. :thinking: 

I think waiting loop will need variety snooze seconds which depends on its condition, so this should be fine to use plain integer inside.

In here, we should check the while loop exited without break, then the `image_conds` might be `[]`, the the code would becomes:
```python
endtime = datetime.now() + timedelta(seconds=wait_timeout)
while endtime > datetime.now():
    code, data = api_client.images.get(unique_name)
    image_conds = data.get('status', {}).get('conditions', [])
    if len(image_conds) > 0:
        break
    sleep(3)
assert image_conds > 0, "Failed to get image ..."
```
equals to
```python
endtime = datetime.now() + timedelta(seconds=wait_timeout)
while endtime > datetime.now():
    code, data = api_client.images.get(unique_name)
    image_conds = data.get('status', {}).get('conditions', [])
    if len(image_conds) > 0:
        break
    sleep(3)
else:
    raise AssertionError(f"Failed to get image, with wait {wait_timeout}s timed out")
```

Reply:
I was just thinking to have some standard variable which is used across the tests. So, if we have any performance change in getting result, we'll get to know from our test.
I'm fine with hardcoded also, this was just a nit.

Reply:
I'm adding in a new fixture with a sleep timeout how we have a wait-timeout. It will pull from the config the same way.